### ISSUE TITLE
UnicodeDecodeError in importer album directory walking

### ISSUE THREAD

**Issue by k-f-** [sampsyo/beets#1285]
I can't be more helpful with this one atm.  Importing over some files I tried to recover from a corrupted file system. Just an error raise that totally stops beets consistently.

If I manage to narrow it down to which file(s) are offensive I'll provide them.

```
Traceback (most recent call last):
  File "/usr/local/bin/beet", line 9, in <module>
    load_entry_point('beets==1.3.11', 'console_scripts', 'beet')()
  File "/usr/local/lib/python2.7/site-packages/beets/ui/__init__.py", line 927, in main
    _raw_main(args)
  File "/usr/local/lib/python2.7/site-packages/beets/ui/__init__.py", line 917, in _raw_main
    subcommand.func(lib, suboptions, subargs)
  File "/usr/local/lib/python2.7/site-packages/beets/ui/commands.py", line 867, in import_func
    import_files(lib, paths, query)
  File "/usr/local/lib/python2.7/site-packages/beets/ui/commands.py", line 844, in import_files
    session.run()
  File "/usr/local/lib/python2.7/site-packages/beets/importer.py", line 316, in run
    pl.run_parallel(QUEUE_SIZE)
  File "/usr/local/lib/python2.7/site-packages/beets/util/pipeline.py", line 251, in run
    msg = self.coro.next()
  File "/usr/local/lib/python2.7/site-packages/beets/importer.py", line 1170, in read_tasks
    for t in task_factory.tasks():
  File "/usr/local/lib/python2.7/site-packages/beets/importer.py", line 1006, in tasks
    for dirs, paths in self.paths():
  File "/usr/local/lib/python2.7/site-packages/beets/importer.py", line 1058, in paths
    for dirs, paths in albums_in_dir(self.toppath):
  File "/usr/local/lib/python2.7/site-packages/beets/importer.py", line 1482, in albums_in_dir
    r'^%s\d' % re.escape(match.group(1)), re.I
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 20: ordinal not in range(128)
```

**Comment by sampsyo** [comment]
Thanks for reporting. I'm guessing that the recent `unicode_literals` change (cc @brunal) might have caused the regex matching here to mix bytes (from paths) with Unicode patterns. But I'll need to construct a test case to be certain—I'll try to get to this soon.

**Comment by brunal** [comment]
I believe changing the regexes from `r''` to `b''` fixes this (there doesn't seem to be a need for raw string anyway). This happens in 2 places. This was not picked up by the test because they use unicode strings with ASCII-only characters.

I pushed a fix on my fork. The test update was a bit abrupt so I did not push it on master -- I want your OK first @sampsyo, cherry-pick the commit if it suits you.
Other tests of the same test suite could receive the same treatment.

**Comment by sampsyo** [comment]
Thanks for taking the first steps, @brunal.

I opted to divide the original tests from the new Unicode tests. This kind of test orthogonality can help isolate the root cause faster if something goes wrong in the future. I also had to wrangle with some fun Unicode normalization problems that arise when we generate filenames for test fixtures.

I also used `br''` literals for the regular expressions. Having those raw literals is helpful if the regexes evolve to be more complex in the future (and they also help indicate to me when I'm reading code that I'm looking at a regex).
